pkg/vere/io/conn.c

```diff
@@ -693,55 +693,79 @@ _conn_init_sock(u3_shan* san_u)
   u3l_log("conn: listening on %s", pip_c);
 
 #else   //  _WIN32
-  //  the full socket path is limited to about 108 characters,
-  //  and we want it to be relative to the pier. save our current
-  //  path, chdir to the pier, open the socket at the desired
-  //  path, then chdir back. hopefully there aren't any threads.
-  //
-  c3_c pax_c[2048];
-  c3_i err_i;
-
-  if ( NULL == getcwd(pax_c, sizeof(pax_c)) ) {
-    u3l_log("conn: getcwd: %s", uv_strerror(errno));
-    u3_king_bail();
+  c3_c  tad_c[U3_SOCK_MAX];
+  c3_w  len_w;
+  c3_i  err_i, fid_i;
+  c3_c* ven_c;
+
+  if ( (ven_c = getenv("TMPDIR")) ) {
+    if ( !(len_w = strlen(ven_c)) ) {
+      ven_c = 0;
+    } else {
+      if ( ven_c[len_w - 1] == '/' ) {
+        --len_w;
+      }
+      if ( len_w >= U3_SOCK_MAX - sizeof("/urbit-XXXXXX/conn.sock") ) {
+        u3l_log("conn: TMPDIR too long");
+        ven_c = 0;
+      }
+    }
   }
-  if ( 0 != chdir(u3_Host.dir_c) ) {
-    u3l_log("conn: chdir: %s", uv_strerror(errno));
-    u3_king_bail();
+  if ( !ven_c ) {
+    ven_c = "/tmp";
+    len_w = 4;
   }
-  if ( 0 != unlink(URB_SOCK_PATH) && errno != ENOENT ) {
-    u3l_log("conn: unlink: %s", uv_strerror(errno));
-    goto _conn_sock_err_chdir;
+  memcpy(tad_c, ven_c, len_w);
+  memcpy(tad_c + len_w, "/urbit-XXXXXX", sizeof("/urbit-XXXXXX"));
+  if ( !mkdtemp(tad_c) ) {
+    u3l_log("conn: mkdtemp: %s", strerror(errno));
+    u3_king_bail();
   }
+  memcpy(tad_c + len_w + sizeof("/urbit-XXXXXX") - 1, "/conn.sock",
```

Comment:
The string "/urbit-XXXXXX" gets reused a lot, we should make it a constant or macro